lib/sw_indexer_engine.rb

```diff
@@ -12,12 +12,15 @@ def index(druid, targets = nil)
 
     ##
     # When a target is not in SKIP_CATKEY_CHECK and its true, check the catkey
-    # If there is a catkey, set false so that the target gets a delete message.
+    # If there is a catkey, set false so that the target gets a delete message and call the 856 marc generation service.
     # We do this so that the "dor" index doesn't get docs indexed that are
     # already coming from the marc record.
     targets.each do |target_key, target_value|
       next unless target_value == true && !Settings.SKIP_CATKEY_CHECK.include?(target_key)
-      targets[target_key] = false if purl_model(druid).catkey.present?
+      if purl_model(druid).catkey.present?
+        targets[target_key] = false
+        update_marc_record(druid)
```

Comment:
I think we want to do this outside the `targets` loop, right? If we have multiple targets, I don't think we need to send multiple messages to the 856 service.